File: sample.ts
/**
 * AutoRepo demo - TypeScript (~70 lines)
 */
function fib(n: number): number[] {
  let a = 0, b = 1;
  const out: number[] = [];
  for (let i = 0; i < n; i++) {
    out.push(a);
    [a, b] = [b, a + b];
  }
  return out;
}

interface Stats { len: number; sum: number; avg: number; max: number; }

function stats(arr: number[]): Stats {
  const sum = arr.reduce((s, x) => s + x, 0);
  return { len: arr.length, sum, avg: arr.length ? sum / arr.length : 0, max: Math.max(...arr) };
}

class Analyzer {
  data: number[];
  constructor(data: number[]) { this.data = data; }
  primes(): number[] { return this.data.filter(n => {
      if (n < 2) return false;
      for (let i = 2; i * i <= n; i++) if (n % i === 0) return false;
      return true;
  }); }
  summary(): Stats { return stats(this.data); }
}

const seq = fib(35);
const a = new Analyzer(seq);
console.log("TS summary:", a.summary());
console.log("TS done.");

File: sample.py
#!/usr/bin/env python3
"""
AutoRepo demo - Python (~70 lines)
Demonstrates functions, classes, and small utilities.
"""
from typing import List, Dict

def fib(n: int) -> List[int]:
    a, b = 0, 1
    out = []
    for _ in range(n):
        out.append(a)
        a, b = b, a + b
    return out

def stats(arr: List[int]) -> Dict[str, float]:
    total = sum(arr)
    length = len(arr)
    avg = total / length if length else 0
    maximum = max(arr) if arr else 0
    return {"len": length, "sum": total, "avg": avg, "max": maximum}

class Analyzer:
    def __init__(self, data: List[int]):
        self.data = data

    def primes(self):
        out = []
        for x in self.data:
            if x > 1 and all(x % i for i in range(2, int(x**0.5)+1)):
                out.append(x)
        return out

    def summary(self):
        return stats(self.data)

if __name__ == "__main__":
    seq = fib(40)
    anal = Analyzer(seq)
    print("Summary:", anal.summary())
    print("Primes:", anal.primes())
    print("Done.")
